core/src/autogluon/core/scheduler/mo_hyperband.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import threading
+import numpy as np
+import multiprocessing as mp
+
+from .hyperband import HyperbandScheduler
+from .reporter import MODistStatusReporter
+from ..utils.default_arguments import filter_by_key
+
+__all__ = ['MOHyperbandScheduler']
+
+logger = logging.getLogger(__name__)
+
+_ARGUMENT_KEYS = {
+    'objectives', 'scalarization_options'
+}
+
+
+class MOHyperbandScheduler(HyperbandScheduler):
+    r"""Implements different variants of asynchronous multi-objective Hyperband
+    by extending the standard HyperbandScheduler using random scalarizations. 
+    The implemented method is described in detail in:
+
+        Schmucker, Donini, Perrone, Zafar, Archambeau (2020)
+        Multi-Objective Multi-Fidelity Hyperparameter Optimization with 
+        Application to Fairness
+        https://meta-learn.github.io/2020/papers/24_paper.pdf
+
+    This class supports the full range of parameters of the standard
+    HyperbandScheduler. The 'objectives' replaces the 'reward_attr' parameter.
+    Here we only explain the multi-objective specific parameters. For the full 
+    list of available options refer to HyperbandScheduler.
+
+    Parameters
+    ----------
+    objectives : dict
```

Comment:
Would it be a big issue if we made all criteria to be maximized?

Reply:
Assuming that all criteria are to be maximized would make the implementation slightly simpler, but it would burden the user with manually flipping the signs of the objective values. Certain objectives such as _computational cost_ and _run time_ are desired to be minimized. We could allow a _list_ input as an alternative way to specify the objectives. If a list is used one would assume that all objectives are intended to be maximized.

Reply:
Makes sense